ocaml/testsuite/tests/typing-unique/overwriting.ml

```diff
@@ -721,12 +774,12 @@ Error: This form is not allowed as the type of the inlined record could escape.
 |}]
 
 let update = function
-  | (Con x) ->
-    let x = overwrite_ x with { x = "foo" } in
-    x
+  | (Con c) ->
+    let x = overwrite_ c with { x = "foo" } in
+    Con x
 [%%expect{|
 Line 3, characters 23-24:
-3 |     let x = overwrite_ x with { x = "foo" } in
+3 |     let x = overwrite_ c with { x = "foo" } in
```

Comment:
Ah, I am not sure I would even leave a CR, because the compiler never allows binding inlined records to variables. For example:
```
let update = function
  | (Con c) -> let x = c in Con x
[%%expect{|
Line 2, characters 23-24:
2 |   | (Con c) -> let x = c in Con x
                           ^
Error: This form is not allowed as the type of the inlined record could escape.
|}]
```
So this would be a larger change in the compiler that actually is not specific to overwriting.

Reply:
Yikes, OK. Sorry for the noise then. The feature is more restrictive than I thought.